src/gapc.cc

```diff
@@ -265,6 +270,33 @@ static void parse_options(int argc, char **argv, Options *rec) {
     rec->plot_grammar_file = basename(rec->out_file) + ".dot";
   }
 
+  if (vm.count("checkpoint")) {
+    std::string cp_interval = vm["checkpoint"].as<std::string>();
+    std::stringstream tmp_interval(cp_interval);
+    std::string val;
+    std::vector<int> interval_vals;
+
+    try {
+      // parse the checkpointing interval the user specified
+      while (std::getline(tmp_interval, val, ':')) {
+        // split the interval string at ':' and store values
+        interval_vals.push_back(std::stoi(val));
+      }
+
+      if (interval_vals.size() != 4) {
+        throw LogError("Invalid checkpointing interval format! "
+                       "Must be \"dd:mm:hh::ss\".");
+      }
+
+      // calculate the interval length (in seconds)
+      int interval = interval_vals[0] * 86400 + interval_vals[1] * 3600 +
+                     interval_vals[2] * 60 + interval_vals[3];
```

Comment:
correct, but isn't there a predefined function that does this trick?

Reply:
`std::chrono` probably contains some function that can do this. Will take a look.

Reply:
> `std::chrono` probably contains some function that can do this. Will take a look.

I was unable to find a function that does exactly what we need here. I would suggest that we leave the interval length calculation as it is (for now).

Reply:
ok